### PR TITLE
Add wxWidgets

### PR DESCRIPTION
**Pull request by hmaarrfk**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [ ] Source is from official source
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Build number is 0
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by hmaarrfk** [comment]
@conda-forge/help-c-cpp I am having a really hard time getting the opengl dependency right.

Any tips?

**Comment by nehaljwani** [comment]
@hmaarrfk Apply this diff:
```diff
diff --git a/recipes/wxwidgets/build.sh b/recipes/wxwidgets/build.sh
index e8a7b1447..951793b13 100644
--- a/recipes/wxwidgets/build.sh
+++ b/recipes/wxwidgets/build.sh
@@ -1,3 +1,5 @@
+export LDFLAGS="${LDFLAGS:-} -fuse-ld=gold -Wl,--as-needed"
+
 ./configure \
   --prefix=${PREFIX} \
   --with-gtk="3" \
```

The `-Wl,--as-needed` is optional, but I would recommend adding anyway. Using gold linker will let you bypass the trouble you are facing right now.

**Comment by hmaarrfk** [comment]
thanks! lets try

**Comment by hmaarrfk** [comment]
seem like the gold linker didn't help :(

**Comment by nehaljwani** [comment]
Could you revert the gold linker change and try this? It passed the CI here: https://github.com/nehaljwani/staged-recipes/pull/4
```diff
diff --git a/recipes/wxwidgets/meta.yaml b/recipes/wxwidgets/meta.yaml
index 51a364612..ad1bc0b0f 100644
--- a/recipes/wxwidgets/meta.yaml
+++ b/recipes/wxwidgets/meta.yaml
@@ -27,6 +27,7 @@ requirements:
     - {{ cdt('libxdamage-devel') }}  # [linux]
     - {{ cdt('libxext-devel') }}  # [linux]
     - {{ cdt('libxi-devel') }}  # [linux]
+    - {{ cdt('libxxf86vm-devel') }}  # [linux]
     - {{ cdt('libxinerama-devel') }}  # [linux]
     - {{ cdt('libxfixes-devel') }}  # [linux]
     - {{ cdt('libxrandr-devel') }}  # [linux]
```

**Comment by nehaljwani** [comment]
I opened an issue at https://github.com/conda-forge/crosstool-ng-feedstock/issues/12 for the gold linker problem (you can still use my second patch to make it work with the bfd linker)

File: recipes/wxwidgets/build.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Use Gold linker and -Wl,--as-needed as suggested in
+# https://github.com/conda-forge/staged-recipes/pull/11589#issuecomment-629882668
+export LDFLAGS="${LDFLAGS:-} -fuse-ld=gold -Wl,--as-needed"
```
Comment (hmaarrfk) [inline recipes/wxwidgets/build.sh:3]:
```suggestion
# export LDFLAGS="${LDFLAGS:-} -fuse-ld=gold -Wl,--as-needed"
```

File: recipes/wxwidgets/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - autotools_clang_conda        # [win]
+    - make
+    - pkg-config
+    - {{ cdt('libice-devel') }}  # [linux]
+    - {{ cdt('libsm-devel') }}  # [linux]
+    - {{ cdt('libx11-devel') }}  # [linux]
```
Comment (hmaarrfk) [inline recipes/wxwidgets/meta.yaml:24]:
```suggestion
    - {{ cdt('libxxf86vm-devel') }}  # [linux]
    - {{ cdt('libx11-devel') }}  # [linux]
```

**Comment by hmaarrfk** [comment]
Thanks for running it on the CIs!

If you had permission, you could have simply pushed to my branch.

Do note that I skipped osx and windows builds.

**Comment by hmaarrfk** [comment]
I'll work on reverting the addition of so many Cdts. I kinda took the shotgun approach and just tried many of them. How did you know it was the one you pointed to?

**Comment by nehaljwani** [comment]
In my CI runs, I had used:
```diff
 ./configure \
   --prefix=${PREFIX} \
   --with-gtk="3" \
-  --with-opengl
+  --with-opengl || (cat config.log && exit -1)
```
That gave me output for whenever the build failed and I inspected it to find out what was missing.

**Comment by hmaarrfk** [comment]
nice trick.

**Comment by hmaarrfk** [comment]
i'll have to start the OSX and windows builds later on this week.

**Comment by hmaarrfk** [comment]
@nehaljwani any clue on how to get OSX building with opengl?

Honestly, I don't care much for OSX, so I'm good with this as is.

**Comment by scopatz** [comment]
Nice work here!

**Pull request merged**